core-tests/shared/src/test/scala/zio/RefSpec.scala

```diff
@@ -45,29 +45,103 @@ object RefSpec
         testM("modify") {
           for {
             ref   <- Ref.make(current)
-            r     <- ref.modify[String](_ => ("hello", update))
+            r     <- ref.modify[Nothing, String](_ => ("hello", update))
```

Comment:
What happens if you omit _both_ type parameters? Does it work?

Reply:
Yes !